python/revng/restapi/producer_data_loader.py

```diff
@@ -0,0 +1,30 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+from aiodataloader import DataLoader
+
+from revng.api import Manager
+
+
+class ProduceDataLoader(DataLoader):
+    def __init__(self, manager: Manager, step: str, container: str):
+        super().__init__()
+        self.produce = lambda x: manager.produce_target(x, step, container)
+
+    async def batch_load_fn(self, keys):
+        return self.produce(keys)
+
+
+class ProducerDataLoader:
+    def __init__(self):
+        self.cache = {}
+
+    def get(self, manager: Manager, step: str, container: str):
+        key = (step, container)
+        if key in self.cache:
+            return self.cache[key]
+        else:
+            loader = ProduceDataLoader(manager, step, container)
+            self.cache[key] = loader
+            return loader
```

Comment:
Please thoroughly comment these two classes, how they work is not obvious.

Reply:
Removed the file, to be re-integrated on the next PR